### PR TITLE
[WIP] Rya-104 for 2017 mongo admin shell

### PR DESCRIPTION
**Pull request by DLotts**
## Description
MongoDB in the Rya Admin Shell

### Tests
Yes, nice coverage so far.

### Links
[Jira](https://issues.apache.org/jira/browse/RYA-104)

### Checklist
- [ ] Code Review
- [ ] Squash Commits

### CODE REVIEW

**Comment by asfgit** [comment]
FAILURE

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/554/<h2>Build result: FAILURE</span></h2>[...truncated 2.06 MB...] at [Source: {"url":"https://api.github.com/repos/apache/incubator-rya/statuses/42e6fed19fc08664ad787fc422d4f9fe479c911c","id":4325905151,"state":"failure","description":"FAILURE\n ","target_url":"https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/554/","context":"Jenkins: clean package -Pgeoindexing","created_at":"2017-11-17T21:55:57Z","updated_at":"2017-11-17T21:55:57Z","creator":{"login":"asfgit","id":1341245,"avatar_url":"https://avatars1.githubusercontent.com/u/1341245?v=4","gravatar_id":"","url":"https://api.github.com/users/asfgit","html_url":"https://github.com/asfgit","followers_url":"https://api.github.com/users/asfgit/followers","following_url":"https://api.github.com/users/asfgit/following{/other_user}","gists_url":"https://api.github.com/users/asfgit/gists{/gist_id}","starred_url":"https://api.github.com/users/asfgit/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/asfgit/subscriptions","organizations_url":"https://api.github.com/users/asfgit/orgs","repos_url":"https://api.github.com/users/asfgit/repos","events_url":"https://api.github.com/users/asfgit/events{/privacy}","received_events_url":"https://api.github.com/users/asfgit/received_events","type":"User","site_admin":false}}; line: 1, column: 125] at [Source: {"url":"https://api.github.com/repos/apache/incubator-rya/statuses/42e6fed19fc08664ad787fc422d4f9fe479c911c","id":4325905151,"state":"failure","description":"FAILURE\n ","target_url":"https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/554/","context":"Jenkins: clean package -Pgeoindexing","created_at":"2017-11-17T21:55:57Z","updated_at":"2017-11-17T21:55:57Z","creator":{"login":"asfgit","id":1341245,"avatar_url":"https://avatars1.githubusercontent.com/u/1341245?v=4","gravatar_id":"","url":"https://api.github.com/users/asfgit","html_url":"https://github.com/asfgit","followers_url":"https://api.github.com/users/asfgit/followers","following_url":"https://api.github.com/users/asfgit/following{/other_user}","gists_url":"https://api.github.com/users/asfgit/gists{/gist_id}","starred_url":"https://api.github.com/users/asfgit/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/asfgit/subscriptions","organizations_url":"https://api.github.com/users/asfgit/orgs","repos_url":"https://api.github.com/users/asfgit/repos","events_url":"https://api.github.com/users/asfgit/events{/privacy}","received_events_url":"https://api.github.com/users/asfgit/received_events","type":"User","site_admin":false}}; line: 1, column: 115] (through reference chain: org.kohsuke.github.GHCommitStatus["id"])	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:388)	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:348)	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1607)	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:278)	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:140)	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3798)	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2842)	at org.kohsuke.github.Requester.parse(Requester.java:616)	... 13 moreCaused by: com.fasterxml.jackson.core.JsonParseException: Numeric value (4325905151) out of range of int at [Source: {"url":"https://api.github.com/repos/apache/incubator-rya/statuses/42e6fed19fc08664ad787fc422d4f9fe479c911c","id":4325905151,"state":"failure","description":"FAILURE\n ","target_url":"https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/554/","context":"Jenkins: clean package -Pgeoindexing","created_at":"2017-11-17T21:55:57Z","updated_at":"2017-11-17T21:55:57Z","creator":{"login":"asfgit","id":1341245,"avatar_url":"https://avatars1.githubusercontent.com/u/1341245?v=4","gravatar_id":"","url":"https://api.github.com/users/asfgit","html_url":"https://github.com/asfgit","followers_url":"https://api.github.com/users/asfgit/followers","following_url":"https://api.github.com/users/asfgit/following{/other_user}","gists_url":"https://api.github.com/users/asfgit/gists{/gist_id}","starred_url":"https://api.github.com/users/asfgit/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/asfgit/subscriptions","organizations_url":"https://api.github.com/users/asfgit/orgs","repos_url":"https://api.github.com/users/asfgit/repos","events_url":"https://api.github.com/users/asfgit/events{/privacy}","received_events_url":"https://api.github.com/users/asfgit/received_events","type":"User","site_admin":false}}; line: 1, column: 125]	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1702)	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:558)	at com.fasterxml.jackson.core.base.ParserBase.convertNumberToInt(ParserBase.java:928)	at com.fasterxml.jackson.core.base.ParserBase._parseIntValue(ParserBase.java:866)	at com.fasterxml.jackson.core.base.ParserBase.getIntValue(ParserBase.java:694)	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:306)	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:286)	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:504)	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:108)	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:276)	... 17 more

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/563/<h2>Build result: FAILURE</span></h2>[...truncated 381.63 KB...][INFO] Apache Rya Spark Support ........................... SKIPPED[INFO] Apache Rya Web Projects ............................ SKIPPED[INFO] Apache Rya Web Implementation ...................... SKIPPED[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 08:38 min[INFO] Finished at: 2017-12-13T22:46:10+00:00[INFO] Final Memory: 146M/3023M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.11:check (check-licenses) on project rya.indexing: Too many files with unapproved license: 2 See RAT report in: /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests/extras/indexing/target/rat.txt -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.indexingchannel stoppedSetting status of 3be4c181f7f8ca39f687a046a40d64abc1dd6751 to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/563/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/564/<h2>Build result: FAILURE</span></h2>[...truncated 381.51 KB...][INFO] Apache Rya Web Projects ............................ SKIPPED[INFO] Apache Rya Web Implementation ...................... SKIPPED[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 05:59 min[INFO] Finished at: 2017-12-14T15:41:42+00:00[INFO] Final Memory: 249M/3139M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.11:check (check-licenses) on project rya.indexing: Too many files with unapproved license: 2 See RAT report in: /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests/extras/indexing/target/rat.txt -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.indexingchannel stoppedSetting status of a2be7833f392d840c1c0483d4c4a85588c9129fc to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/564/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/565/<h2>Build result: FAILURE</span></h2>[...truncated 637.58 KB...][INFO] Apache Rya Spark Support ........................... SKIPPED[INFO] Apache Rya Web Projects ............................ SKIPPED[INFO] Apache Rya Web Implementation ...................... SKIPPED[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 13:56 min[INFO] Finished at: 2017-12-15T22:35:35+00:00[INFO] Final Memory: 415M/3064M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.11:check (check-licenses) on project rya.indexing: Too many files with unapproved license: 2 See RAT report in: /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests/extras/indexing/target/rat.txt -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.indexingchannel stoppedSetting status of 72d90f9a2850fc38f7d0d5dd53edcbadffb3e2e3 to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/565/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

File: extras/indexing/src/test/java/org/apache/rya/api/client/mongo/MongoLoadStatementsFileIT.java [hunk 1]
```diff
@@ -18,97 +18,114 @@
...
+    public void instanceDoesNotExist() throws Exception {
+
+        final RyaClient ryaClient = MongoRyaClientFactory.build(getConnectionDetails(), conf.getMongoClient());
+        ryaClient.getLoadStatementsFile().loadStatements(getConnectionDetails().getHostname(), Paths.get("src/test/resources/example.ttl"), RDFFormat.TURTLE);
+    }
+
+    @Test
+    public void loadTurtleFile() throws Exception {
+        // Install an instance of Rya.
+        final InstallConfiguration installConfig = InstallConfiguration.builder()//
```
Comment (kchilton2) [inline extras/indexing/src/test/java/org/apache/rya/api/client/mongo/MongoLoadStatementsFileIT.java]: Why do you put // on the ends of your lines? That's completely unnecessary.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/568/<h2>Build result: FAILURE</span></h2>[...truncated 1.85 MB...][INFO] ------------------------------------------------------------------------[INFO] Total time: 39:23 min[INFO] Finished at: 2017-12-19T21:25:25+00:00[INFO] Final Memory: 197M/2672M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.2:compile (default-compile) on project rya.shell: Compilation failure[ERROR] /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests@2/extras/shell/src/main/java/org/apache/rya/shell/RyaConnectionCommands.java:[194,62] constructor MongoConnectionDetails in class org.apache.rya.api.client.mongo.MongoConnectionDetails cannot be applied to given types;[ERROR] required: java.lang.String,char[],java.lang.String,int,java.lang.String[ERROR] found: java.lang.String,char[],java.lang.String,int[ERROR] reason: actual and formal argument lists differ in length[ERROR] -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.shellchannel stoppedSetting status of a75bc1c6faef3a5425b0552119960191b3762c0d to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/568/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/567/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya' /><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/567/org.apache.rya$mongodb.rya/testReport'>incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/567/org.apache.rya$mongodb.rya/testReport/org.apache.rya.mongodb.instance/MongoDetailsAdapterTest/absentOptionalToRyaDetailsTest/'><strong>org.apache.rya.mongodb.instance.MongoDetailsAdapterTest.absentOptionalToRyaDetailsTest</strong></a></li></ul>

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/569/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya' /><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/569/org.apache.rya$mongodb.rya/testReport'>incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/569/org.apache.rya$mongodb.rya/testReport/org.apache.rya.mongodb.instance/MongoDetailsAdapterTest/absentOptionalToRyaDetailsTest/'><strong>org.apache.rya.mongodb.instance.MongoDetailsAdapterTest.absentOptionalToRyaDetailsTest</strong></a></li></ul>

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/570/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya' /><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/570/org.apache.rya$mongodb.rya/testReport'>incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/570/org.apache.rya$mongodb.rya/testReport/org.apache.rya.mongodb.instance/MongoDetailsAdapterTest/absentOptionalToRyaDetailsTest/'><strong>org.apache.rya.mongodb.instance.MongoDetailsAdapterTest.absentOptionalToRyaDetailsTest</strong></a></li></ul>

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/573/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya' /><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/573/org.apache.rya$mongodb.rya/testReport'>incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/573/org.apache.rya$mongodb.rya/testReport/org.apache.rya.mongodb.instance/MongoDetailsAdapterTest/absentOptionalToRyaDetailsTest/'><strong>org.apache.rya.mongodb.instance.MongoDetailsAdapterTest.absentOptionalToRyaDetailsTest</strong></a></li></ul>

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/574/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya' /><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/574/org.apache.rya$mongodb.rya/testReport'>incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/574/org.apache.rya$mongodb.rya/testReport/org.apache.rya.mongodb.instance/MongoDetailsAdapterTest/absentOptionalToRyaDetailsTest/'><strong>org.apache.rya.mongodb.instance.MongoDetailsAdapterTest.absentOptionalToRyaDetailsTest</strong></a></li></ul>

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/576/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/577/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya' /><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/577/org.apache.rya$mongodb.rya/testReport'>incubator-rya-master-with-optionals-pull-requests/org.apache.rya:mongodb.rya</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/577/org.apache.rya$mongodb.rya/testReport/org.apache.rya.mongodb.instance/MongoDetailsAdapterTest/absentOptionalToRyaDetailsTest/'><strong>org.apache.rya.mongodb.instance.MongoDetailsAdapterTest.absentOptionalToRyaDetailsTest</strong></a></li></ul>

**Pull request closed without merging**